### DISCUSSION TITLE
Local Development - Using Extensions

### DISCUSSION THREAD

**Discussion by rkho** [supabase/supabase#2250] [Questions]
Hey all I'm trying to use a db dump that I created off Supabase with the local Supabase dev environment but I'm running into some issues with the `uuid_generate_v4` extension

I'm initializing with `CREATE EXTENSION IF NOT EXISTS "uuid-ossp";`

Here's where it fails:

`select_uuid uuid DEFAULT extensions.uuid_generate_v4() NOT NULL,`

And the error message:

```
psql:db_dump.sql:57: ERROR:  function extensions.uuid_generate_v4() does not exist
LINE 2:     select_uuid uuid DEFAULT extensions.uuid_generate_v4() N...
```

What might I be doing wrong? I've been looking through Stack for a bit and haven't found any hints. Anything that could help unstick me would be great!

**Comment by steve-chavez** [comment]
Hey @rkho,

Try adding a schema to the extension:

```sql
-- CREATE SCHEMA IF NO EXISTS extensions; (optional)
CREATE EXTENSION IF NOT EXISTS "uuid-ossp" SCHEMA extensions;
```

**Reply by rkho** [reply]
Thanks for the followup;

I modified my script as follows:

```

CREATE SCHEMA IF NOT EXISTS extensions;

CREATE EXTENSION IF NOT EXISTS "uuid-ossp" SCHEMA extensions;
```

I'm still receiving the same issue:

```
psql:db_dump.sql:57: ERROR:  function extensions.uuid_generate_v4() does not exist
LINE 2:     select_uuid uuid DEFAULT extensions.uuid_generate_v4() N...

HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
```

**Reply by sroussey** [reply]
Same problem here (using local dev install on docker). 

```postgres=# CREATE EXTENSION IF NOT EXISTS "uuid-ossp" SCHEMA extensions;
NOTICE:  extension "uuid-ossp" already exists, skipping
CREATE EXTENSION
postgres=# \df
                            List of functions
 Schema |      Name       | Result data type | Argument data types | Type 
--------+-----------------+------------------+---------------------+------
 auth   | email           | text             |                     | func
 auth   | role            | text             |                     | func
 auth   | uid             | uuid             |                     | func
 public | handle_new_user | trigger          |                     | func
(4 rows)
```

**Reply by devon2018** [reply]
Having the same issue with local docker install

**Reply by michaeldever** [reply]
Same here

**Reply by ferpintado** [reply]
Having the same issue,  trying to integrate prisma migrate with supabase

**Comment by clarepure** [comment]
https://stackoverflow.com/questions/22446478/extension-exists-but-uuid-generate-v4-fails

This worked for me
